### PR TITLE
Various changes for SSA agent running inside docker

### PR DESCRIPTION
**Pull request by hsong-rh**
Changes to enable SSA agent to read disks information either from docker or from EC2 instance. 

This PR depends on https://github.com/ManageIQ/manageiq-providers-amazon/pull/318.

### CODE REVIEW

**Comment by hsong-rh** [comment]
@roliveri @jerryk55 @Fryguy Please review.

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_instance.rb [hunk 1]
```diff
@@ -14,16 +13,22 @@ def unmap_volumes
       super
       while (snap = @snapshots.shift)
         snap.delete
+        _log.info("Snapshot #{snap.id} is deleted!")
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_instance.rb]: No need for "is", just say:
```ruby
_log.info("Snapshot #{snap.id} deleted")
```

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_instance.rb [hunk 2]
```diff
@@ -14,16 +13,22 @@ def unmap_volumes
...
-      _log.debug("    Creating snapshot of instance volume #{id}")
-      snap = @ec2.create_snapshot(volume_id: id, description: "MIQ extract snapshot for instance: #{@ec2_obj.id}")
+    def create_snapshot(vol_id)
+      _log.info("    Creating snapshot of instance volume #{vol_id}")
+      snap = @ec2.create_snapshot(volume_id: vol_id, description: "SSA extract snapshot for instance: #{@ec2_obj.id}")
       snap.wait_until_completed
-      snap.create_tags(tags: [{key: 'Name', value: 'MIQ extract snapshot'}])
-      _log.debug("    Creating snapshot of instance volume #{id} DONE snap_id = #{snap.id}")
+      snap.create_tags(tags: [{key: 'Name', value: 'SSA extract snapshot'}])
+      _log.info("    Snapshot #{snap.id} of instance volume #{vol_id} is created!")
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_instance.rb]: Here too, no need for "is":
```ruby
_log.info("    Snapshot #{snap.id} of instance volume #{vol_id} created")
```

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb [hunk 3]
```diff
@@ -96,26 +70,34 @@ def create_volume(snap_id)
       sleep 2
       @ec2.client.wait_until(:volume_available, volume_ids: [volume.id])
 
-      volume.create_tags(tags: [{ key: 'Name', value: 'MIQ extract volume'},
-                                { key: 'Description', value: "MIQ extract volume for image: #{@ec2_obj.id}"}])
+      volume.create_tags(tags: [{ key: 'Name', value: 'SSA extract volume'},
+                                { key: 'Description', value: "SSA extract volume for image: #{@ec2_obj.id}"}])
 
-      _log.debug("    Creating volume based on #{snap_id} DONE")
+      _log.info("    Volume #{volume.id} of snapshot #{snap_id} is created!")
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb]: Another unneeded "is":
```ruby
_log.info("    Volume #{volume.id} of snapshot #{snap_id} created")
```

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb [hunk 4]
```diff
@@ -124,16 +106,33 @@ def unmap_volumes
       while (vol = @volumes.shift)
         attachment = vol.detach_from_instance(instance_id: @host_instance.id, force: true)
 
-        _log.debug("#{vol.id} is #{attachment.state}")
+        _log.info("#{vol.id} is #{attachment.state}")
         @ec2.client.wait_until(:volume_available, volume_ids: [vol.id])
         vol.delete
         @ec2.client.wait_until(:volume_deleted, volume_ids: [vol.id])
-        _log.debug("Volume #{vol.inspect} is deleted!")
+        _log.info("Volume #{vol.id} is deleted!")
```
Comment (roliveri) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb]: Again:
```ruby
_log.info("Volume #{vol.id} deleted")
```

File: lib/amazon_ssa_support/ssa_queue_extractor.rb [hunk 5]
```diff
@@ -32,12 +32,12 @@ def extract_loop(timeout)
...
             start = Time.now.to_i # reset time counter after message is processed
             return @exit_code if @exit_code
-            _log.debug("Waiting for next message")
+            _log.info("Waiting for next message")
           end
         end
         break if (Time.now.to_i - start) >= timeout
       end
-      _log.debug("No messages received in #{timeout} seconds, agent shuts down!!!")
+      _log.info("No messages received in #{timeout} seconds, agent shuts down!!!")
```
Comment (roliveri) [inline lib/amazon_ssa_support/ssa_queue_extractor.rb]: Change "shuts" to "shutting":
```ruby
_log.info("No messages received in #{timeout} seconds, agent shutting down")
```

**Comment by hsong-rh** [comment]
@roliveri Changes are made based on your comments. Please review.

File: lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb [hunk 6]
```diff
@@ -124,16 +106,33 @@ def unmap_volumes
...
+
+    def get_first_free_device(device_prefix = '/dev/xvd')
+      all_devices = []
 
-      while (snap = @snapshots.shift)
-        snap.delete
+      # Aws EC2 starts from /dev/xvdf
+      ('f'..'z').each do |char|
+        all_devices << "#{device_prefix}#{char}"
       end
```
Comment (Fryguy) [inline lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_ebs_vmbase.rb]: Minor, but this simplifies to

```ruby
all_devices = ('f'..'z').collect { |c| "#{device_prefix}#{c}" }
```

File: spec/miq_ec2_vm/miq_ec2_ebs_instance_spec.rb [hunk 7]
```diff
@@ -3,6 +3,9 @@
 require_relative '../spec_helper'
 require_relative '../aws_ssa_commons'
 
+::Module.include(LogDecorator::Logging::ClassMethods)
+include LogDecorator::Logging
```
Comment (Fryguy) [inline spec/miq_ec2_vm/miq_ec2_ebs_instance_spec.rb]: I'm surprised you need this in specs,  but even so, I'd think this belongs in spec_helper.

**Pull request merged**